app/src/main/java/com/android/voyageur/ui/overview/AddTrip.kt

```diff
@@ -131,25 +149,35 @@ fun AddTripScreen(
             activities = listOf(),
             type = tripType,
             imageUri = imageUrl)
-
-    tripsViewModel.createTrip(trip)
-    navigationActions.goBack()
+    if (!isEditMode) {
+      tripsViewModel.createTrip(trip)
+      navigationActions.goBack()
+    } else {
+      tripsViewModel.updateTrip(
+          trip,
+          onSuccess = {
+            tripsViewModel.selectTrip(Trip())
+            tripsViewModel.selectTrip(trip)
```

Comment:
Yes, both of these "tricks" are necessary in order to force a recompose, otherwise the UI won't update as it doesn't detect a change in the object references. The repository ins't called in either of these cases so there are no requsts made.